cockpit-actions: Allow auto-running once or on intervals

rafaellehmkuhl opened the pull request:
https://github.com/user-attachments/assets/98ec7444-735f-45ba-9bfb-b23109709528

ArturoManzoli reviewed (commented):
Works very well and looks good too.

One suggestion:
How about making those time inputs in seconds? It will be more human readable. Small intervals are easy to convert, but 2 or 10 minutes, for example, may start adding extra effort to input.
Even if there is a case that the user needs a millisecond delay or interval, it can be set as 0.25 s.
 
![image](https://github.com/user-attachments/assets/d9ff0bc4-71f5-497c-9e42-958fdb5a31bb)

rafaellehmkuhl commented:
> Works very well and looks good too.
> 
> One suggestion: How about making those time inputs in seconds? It will be more human readable. Small intervals are easy to convert, but 2 or 10 minutes, for example, may start adding extra effort to input. Even if there is a case that the user needs a millisecond delay or interval, it can be set as 0.25 s.

Fair. Will change.

rafaellehmkuhl commented:
> Works very well and looks good too.
> 
> One suggestion: How about making those time inputs in seconds? It will be more human readable. Small intervals are easy to convert, but 2 or 10 minutes, for example, may start adding extra effort to input. Even if there is a case that the user needs a millisecond delay or interval, it can be set as 0.25 s.

Done.

ES-Alexander commented on src/components/configuration/ActionLinkConfig.vue:
```diff
@@ -37,18 +34,49 @@
         />
 
+        <div class="mt-4">
+          <p class="text-subtitle-1 font-weight-bold mb-2">Run time-based:</p>
+          <v-radio-group v-model="dialog.autoRunType" class="mt-2">
+            <v-radio value="none" label="Do not run automatically"></v-radio>
+            <v-radio value="once" label="Run automatically once"></v-radio>
+            <v-radio value="interval" label="Run automatically frequently"></v-radio>
+          </v-radio-group>
+
+          <v-text-field
+            v-if="dialog.autoRunType === 'once'"
+            v-model="startupDelaySeconds"
+            label="Delay after startup (s)"
+            type="number"
+            min="0"
+            variant="outlined"
+            density="compact"
+            class="mt-2"
+          />
+
+          <v-text-field
+            v-if="dialog.autoRunType === 'interval'"
+            v-model="runFrequencySeconds"
+            label="Run every (s)"
+            type="number"
+            min="0"
+            variant="outlined"
+            density="compact"
+            class="mt-2"
+          />
```
```suggestion
          <p class="text-subtitle-1 font-weight-bold mb-2">Time-based:</p>
          <v-radio-group v-model="dialog.autoRunType" class="mt-2">
            <v-radio value="once" label="On startup"></v-radio>
            <v-radio value="interval" label="Repeatedly"></v-radio>
            <v-radio value="none" label="No time trigger"></v-radio>
          </v-radio-group>

          <v-text-field
            v-if="dialog.autoRunType !== 'none'"
            v-model="runFrequencySeconds"
            :label="dialog.autoRunType === 'interval' ? 'Run every' :  'Run after'"
            suffix="s"
            type="number"
            min="0"
            variant="outlined"
            density="compact"
            class="mt-2"
          />
```
Given the similarities between the interval and once options, it seems cleaner to combine them and just make the label dynamic. Will require updating the code to use a single variable.

ES-Alexander commented on src/components/configuration/ActionLinkConfig.vue:
```diff
@@ -37,18 +34,49 @@
         <v-text-field
           v-model="dialog.minInterval"
-          label="Minimum interval between calls (ms)"
+          label="Minimum interval between variable-based calls (ms)"
```
Ok, then maybe just relabel as "With minimum gap (ms/change)" for now? :-)

rafaellehmkuhl replied:
Was actually able to implement it!

ArturoManzoli reviewed (approved):
Everything looks fine from my side now. 
I also agree with @ES-Alexander textual changes

rafaellehmkuhl commented:
@ES-Alexander can you take a look at how it got? I was able to move the variable-based calls to be set as a frequency as well (by keeping the stored value as it is, an interval, and changing only the UI, with a get/set computed).

ES-Alexander commented on src/components/configuration/ActionLinkConfig.vue:
```diff
@@ -46,10 +45,24 @@
-            between consecutive calls.
-          </p>
+          <p class="text-subtitle-1 font-weight-bold mb-2">Time-based:</p>
+          <v-radio-group v-model="dialog.autoRunType" class="mt-2">
+            <v-radio value="once" label="On startup"></v-radio>
+            <v-radio value="interval" label="Repeatedly"></v-radio>
+            <v-radio value="none" label="No time trigger"></v-radio>
+          </v-radio-group>
```
Could this be a horizontal row, in order of frequency (e.g. `None | On startup | Repeatedly`)?
That seems more intuitive than the current ordering/placement. 

Ideally it would be a `v-slider`, with the [options as tick labels](https://vuetifyjs.com/en/components/sliders/#ticks), in which case the display condition for the text field could be `frequencySlider > 0`.

rafaellehmkuhl replied:
Can we keep as it is in the current state? The user can slider for easy setting or type what they want for custom values. All edge-cases are already treated.

Also, I personally don't think the slider for the none/once/intervaled is very intuitive, and it also asks for a drag when a click would be enough.

ES-Alexander reviewed (changes requested):
Looking better - nearly there! :D

A couple of suggestions (one missed, one new).
Starting #1811 here would also be nice, but is likely out of scope for this PR, since that component was already in place before the PR.

rafaellehmkuhl commented:
> Looking better - nearly there! :D
> 
> 
> 
> A couple of suggestions (one missed, one new).
> 
> Starting #1811 here would also be nice, but is likely out of scope for this PR, since that component was already in place before the PR.

Will make the changes!

If the autocomplete works without problema, o will add it as well.

rafaellehmkuhl commented:
@ES-Alexander everything implemented:
- Texts changed
- Options order changed
- Replaced the two inputs with a single `v-autocomplete`
- Replaced the text-only interval definer with a slider+text one (user can type smaller or greater values if needed)
